### DISCUSSION TITLE
Something is wrong with gobject-introspection 1.70.0 bottle for Catalina

### DISCUSSION THREAD

**Discussion by peat-psuwit** [Homebrew/discussions#2272] [Everyday usage]
For some reason, gobject-introspection metadata for `g_spawn_command_line_sync()` is incorrect. With `pygobject3` installed, this demonstrates the problem:

```
% python3
Python 3.9.7 (default, Sep  3 2021, 12:45:31) 
[Clang 12.0.0 (clang-1200.0.32.29)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from gi.repository import GLib
>>> GLib.spawn_command_line_sync('echo foo')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: GLib.spawn_command_line_sync() takes exactly 2 arguments (1 given)
>>>
```

However, when re-install from source, the output is correct, and match the output from my Linux machine:

```
% python3
Python 3.9.7 (default, Sep  3 2021, 12:45:31) 
[Clang 12.0.0 (clang-1200.0.32.29)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from gi.repository import GLib
>>> GLib.spawn_command_line_sync('echo foo')
(True, standard_output=b'foo\n', standard_error=b'', wait_status=0)
>>>
```

So, there's something wrong with the building of the bottle. I'm not sure how the bottles are built, so don't know where to debug.

**Comment by SMillerDev** [comment]
Could you try and make a pull request to add the failing command to the test block of the formula? That way we can verify that is a problem in our builds and we'll be able to avoid it in the future.

**Reply by peat-psuwit** [reply]
Hmm, won't this create a circular dependency? I mean, `pygobject` is used during the tests only (and that can be specified in the formula), but I'm not sure how that would work when e.g. bootstraping or something.

Also, digging deeper, it seems like this happens because [GLib's recent argument rename](https://gitlab.gnome.org/GNOME/glib/-/commit/e0b6b8037d49adb5096f1ba584c80e58686a367a) which somehow confuses gobject-introspection [1]. Adding a test would detect this specific breakage, but I'm not sure if it helps with future breakage.

Will try to add one anyway though.

[1] I inspected the Gir file for GLib, and discovered that it describes the renamed argument for that function wrong.
